### PR TITLE
[CSL-1626] Improve DNS support

### PR DESCRIPTION
**Pull request by adinapoli-iohk**
This PR should fix all the woes described here: https://issues.serokell.io/issue/CSL-1626

In a nutshell, this PR:

* Removes all the Windows-specific code for DNS resolutions;
* Bump the `dns` version to the latest on master (`3.0.0` at the time of opening this patch).

The rationale is that proper support for Windows has been merged upstream, which includes not only the work @rockbmb started, but also support for multiple nameservers.

### CODE REVIEW

**Comment by adinapoli-iohk** [comment]
> It's surprising to me though that defaultResolvConf is still pure upstream, even on Windows. Doesn't it need to do a syscall to figure out the name server?

That's because in reality no decision is made when you call `defaultResolvConf` , but rather when you "interpret" it when building a new `ResolvSeed`. At that point the Cthulhu will be unleashed 😛

**Comment by edsko** [comment]
> That's because in reality no decision is made when you call defaultResolvConf , but rather when you "interpret" it when building a new ResolvSeed. At that point the Cthulhu will be unleashed :stuck_out_tongue:

Aha, I see. That makes sense.

**Comment by adinapoli-iohk** [comment]
Ah crap I've forgot to remove this spec 🙈 :

```
C:\w\node\test\Test\Pos\Network\Windows\DnsDomainsSpec.hs:23:1: error:
    Failed to load interface for `Pos.Network.Windows.DnsDomains'
    Perhaps you meant
      Pos.Network.DnsDomains (from cardano-sl-infra-0.6.0)
    Use -v to see a list of the files searched for.
--  While building package cardano-sl-0.6.0 using:
```

File: stack.yaml [hunk 1]
```diff
@@ -81,10 +81,10 @@ packages:
     - engine-io
     - engine-io-wai
 
-# Forked version of "dns" with proper Windows support
+# Forked version of "dns" with proper Windows & Unix multiple-nameservers support
 - location:
-    git: https://github.com/input-output-hk/dns.git
-    commit: 6da3cd051bd6f6415c8ec3b53f1c0372a80a4376
+    git: https://github.com/kazu-yamamoto/dns.git
+    commit: 08df7fe6a9242f7d76aa5040221702e26ab610f6
```
Comment (rockbmb) [inline stack.yaml]: I'm not sure how long it'll take, but it would be good not to forget to remove this after `dns` updates in Hackage.

**Comment by dcoutts** [comment]
The C code could be better, likely space leak and buffer overflow. See https://github.com/kazu-yamamoto/dns/pull/83#pullrequestreview-63847079

**Comment by domenkozar** [comment]
@dcoutts this was merged to 1.0, can we close it?

```
$ git log --grep CSL-1626
commit cd0567e68450ed513ec7b4c1b73054d06ac4f122 (origin/cardano-sl-1.0-cherrypicking)
Author: Alfredo Di Napoli <alfredo.dinapoli@iohk.io>
Date:   Tue Sep 19 11:36:37 2017 +0200

    [CSL-1626] Improve DNS support

    (cherry picked from commit bb6d8668434dac621490226ae9d74c6157cf44cc)

    [CSL-1626] Remove obsolete DnsDomainsSpec

    (cherry picked from commit 986a3503bb39779c9472579b1907cb01b99d16cb)
```

**Comment by adinapoli-iohk** [comment]
@domenkozar We are in the process of furthering refining the DNS support as the C/FFI code could be better:

https://github.com/kazu-yamamoto/dns/pull/84

I would suggest living this PR here as a testament so we don't forget to update it when the improved Windows support will be ready, but your call guys, as I'm easy.

**Comment by adinapoli-iohk** [comment]
@domenkozar @dcoutts Finally we have the latest version of DNS merged on master with the improved FFI code. As this PR accumulated conflicts over time, I'm going to close and reopen.

**Pull request closed without merging**